delegate.od

```diff
@@ -3,57 +3,63 @@
 #
 # Our job is then to delegate the work the right .do file inside $SRCDIR,
 # if any.
-OUT=$PWD
-ofile=$PWD/$3
-x1=$1
-cd "$SRCDIR"
-redo-whichdo "$x1" | {
-    ifcreate=
-    while read dopath; do
-        if [ ! -e "$dopath" ]; then
-            ifcreate="$ifcreate $dopath"
-        else
-            redo-ifcreate $ifcreate
-            redo-ifchange "$dopath"
-
-            # This is slightly faster than forking $(dirname) and $(basename),
-            # plus doesn't try to do any magic like using '.' for an empty
-            # dir field.
-            dofile=${dopath##*/}
-            dodir=${dopath%$dofile}
-
-            # Calculate the $1 path relative to the new $dodir.
-            # Note: if $dodir is nonempty, it will have a trailing slash,
-            # because of the way it's constructed.
-            #
-            # What we want is to run $dofile from $dodir in the *source*
-            # directory.  We then lie a bit in $1/$2: we provide a relative
-            # path to where we expect the *source* file to reside, even though
-            # $3 will actually end up putting the output at the same relative
-            # path inside $OUT.
-            x1_rel=${x1#$dodir}
-
-            # Calculate the $2 relative path.
-            # We can't start with $2 here, because we were called from
-            # default.do and redo-whichdo might have afound a default.o.do,
-            # for example.  We have to strip the extension from $1 ourselves.
-            ext=${dofile##*default}
-            if [ "$ext" != "$dofile" ]; then
-                ext=${ext%.do}
+redo-use-whichdo(){
```

Comment:
In general, I agree that these changes make it quite a bit more readable.  I don't think we can accept pull requests unless you at least try running the code first, though :)

Let me know if you have any trouble making it compile.  it's supposed to be... relatively... easy.  If you have trouble, you can try the https://github.com/apenwarr/wvbuild project, which builds some of the slightly trickier dependencies (like a specific openssl-dev version; we rely on 1.0.x I think).

Reply:
Dumb question: how do I *use* wvbuild? I ran redo and afaict it built fine, but presumably I need to put it in some include path: in If I try to just `cd ~/apenwarr/wvstreams; mkdir out; cd out; ../configure` it eventually complains about
```
configure.real: WARNING: DBUS is missing.
configure.real: WARNING: Qt is missing.
configure.real: WARNING: Valgrind is missing.
configure.real: WARNING: OpenSSL is missing.
configure.real: error: Required dependencies missing: OpenSSL>=0.9.7
```
as it did initially.